### PR TITLE
add completer for pip

### PR DESCRIPTION
**Pull request by Granitas**
Seems like this should be in the core.

This completes commands from `--help` and completes `pip list` values for `pip show` and `pip uninstall`.

### CODE REVIEW

File: xonsh/completers/pip.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import re
+import subprocess
+
+RE_PIP = re.compile("pip(?:\d|\.)*")
+RE_PIP_LIST = re.compile("pip(?:\d|\.)* (?:uninstall|show)")
```
Comment (scopatz) [inline xonsh/completers/pip.py]: Please make these lazy

File: xonsh/completers/pip.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+
+def get_pip_commands():
+    help_text = str(subprocess.check_output(['pip', '--help'], stderr=subprocess.DEVNULL))
+    commands = re.findall("  (\w+)  ", help_text)
+    return [c for c in commands if c not in ['completion', 'help']]
+
+
+ALL_COMMANDS = get_pip_commands()
```
Comment (scopatz) [inline xonsh/completers/pip.py]: same here

**Comment by scopatz** [comment]
Thanks a ton for putting this in!

**Comment by scopatz** [comment]
Yeah that is what I meant :)

**Comment by Granitas** [comment]
For some reason the build on python 3.5 timeout, but everything seems to be fine. Could someone maybe restart it?

**Comment by scopatz** [comment]
Hi @Granitas - I think that this needs a news entry http://xon.sh/devguide.html#changelog and the minor pip ordering change.

File: xonsh/completers/pip.py [hunk 3]
```diff
@@ -22,7 +22,12 @@ def complete_pip(prefix, line, begidx, endidx, ctx):
     """
     Completes python's package manager pip
     """
-    if not PIP_RE.search(line):
+    case = [
+        not PIP_RE.search(line),
+        len(line.split()) > 3,
+        len(line.split()) > 2 and line.endswith(' '),
+    ]
```
Comment (laerus) [inline xonsh/completers/pip.py]: if you do something like this `line_len = len(line.split())` at the start of the function you will create only one list and calculate its length only once. a little more sufficient!

Comment (Granitas) [inline xonsh/completers/pip.py]: Good catch!

**Comment by scopatz** [comment]
Thanks @Granitas!

**Pull request merged**